### PR TITLE
[BEAM-27, BEAM-362] Move InMemoryTimerInternals to runners-core; restore a deprecated copy in its place

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

This is a replay of ec0bf7b4023ff75f4ec6723d2e77ed507eb57c51 with an additional adapter commit that should enable integration tests to succeed.

Note that this configuration of commits works well with the current capabilities of `git` and GitHub.  This behavior is not baked into the history, but is calculated dynamically by both, so it may improve over time.

Notes on what would happen if this were done in one commit:

 - `git log --follow` finds the history
 - `git diff --find-copies` finds the copy
 - `git blame --find-copies-harder` finds the copy, but `--find-copies` does not

I don't recall which level GitHub's UI works with, but the most important two are `git log` and `git blame`. To make sure it works without any of the above, two commits is better.

### CODE REVIEW

**Comment by kennknowles** [comment]
R: @bjchambers do you have time to look at an easy one?

**Comment by kennknowles** [comment]
R: @tgroh -@bchambers because he's out for a bit

File: runners/core-java/src/main/java/org/apache/beam/runners/core/InMemoryTimerInternals.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.apache.beam.sdk.util.TimeDomain;
+import org.apache.beam.sdk.util.TimerInternals;
+import org.apache.beam.sdk.util.WindowTracing;
+import org.apache.beam.sdk.util.state.StateNamespace;
+import org.joda.time.Instant;
+
+/**
+ * Simulates the firing of timers and progression of input and output watermarks for a single
+ * computation and key in a Windmill-like streaming environment.
```
Comment (tgroh) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/InMemoryTimerInternals.java]: Pls remove implementation details.

Comment (kennknowles) [inline runners/core-java/src/main/java/org/apache/beam/runners/core/InMemoryTimerInternals.java]: Done

**Pull request merged**